### PR TITLE
WIP: Move to BinaryProvider build

### PR DESCRIPTION
**Pull request by ianshmean**
This is a VERY WIP start, to help figure out what needs to be done

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,5 +1,10 @@
 *.swp
-deps/deps.jl
-docs/build
```
Comment (remi-garcia) [inline .gitignore]: I think `docs/build` shouldn't be removed

Comment (remi-garcia) [reply .gitignore]: Oh, I shouldn't comment when I'm that tired, sorry

**Comment by tknopp** [comment]
what is with Cairo and so on?, will that be pulled by Cairo.jl?

**Comment by lobingera** [comment]
i'll try to integrate this: https://github.com/giordano/Yggdrasil/releases/tag/Cairo-v1.14.12 and use https://github.com/JuliaGraphics/Cairo.jl/pull/229 (needs rebasing on master and somework on Project.toml). The build.jl for BP is not straightforward to understand imho. Then we should have basic testing with this - without pango/text, Next steps would be to ramp-up something similar for libpango. Then i'd assume we need somethinglike X11 testing of cairo (i worked for some time on XCB - cairo integration but ...). And in the meanwhile the binarybuilder regulars will mostlikely have Gtk builder running.

**Comment by jonathanBieler** [comment]
Is there something I can do to help ? I've got a bit of time but I'm not very familiar with BinaryBuilder  or the current state of affair with all these libraries. Maybe @giordano can give us some tips too.

**Comment by giordano** [comment]
If you want to follow the progress, have a look at this issue: https://github.com/JuliaPackaging/Yggdrasil/issues/62. Most of the direct dependencies should be ready, there are a couple still missing, which have a long chain of indirect dependencies

**Comment by jonathanBieler** [comment]
Looks great, thanks. If you need help with testing or something else, feel free to ping me.

**Comment by tknopp** [comment]
superseded by #447

**Pull request closed without merging**